### PR TITLE
[RFR] Change the puzzle size

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Offer a (W)idth command to change the grid width
- [x] Improve some UI

### CODE REVIEW

**Comment by Luwangel** [comment]
peek 17-10-2017 15-49

File: src/game/exception.py [hunk 1]
```diff
@@ -2,6 +2,10 @@ class GameRuleException(Exception):
     pass
 
 
+class SizeNotValidException(GameRuleException):
```
Comment (Kmaschta) [inline src/game/exception.py]: Size of what?

File: src/game/game.py [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 
 
-def build_grid(size=4):
+MIN_SIZE = 2
+MAX_SIZE = 10
+DEFAULT_SIZE = 4
+
+
+def build_grid(size=DEFAULT_SIZE):
+    if not MIN_SIZE <= size <= MAX_SIZE:
```
Comment (Kmaschta) [inline src/game/game.py:13]: Can you test the upper and lower bound error ?
I don't see this error on the test

File: src/index.py [hunk 3]
```diff
@@ -32,13 +43,35 @@ def play(grid_arg, started_grid):
         grid, shuffled = play_one_turn(grid, shuffled)
 
 
+def change_size(min_size, max_size, default_size):
+    if input(show_menu_size('W')) in ['w', 'W']:
+        size_valid = False
+        while not size_valid:
+            try:
+                size = int(input(ask_size()))
+                if not min_size <= size <= max_size:
```
Comment (Kmaschta) [inline src/index.py]: Duplicated code from build grid. You better have to check for invalid size exception

File: src/renderer/renderer.py [hunk 4]
```diff
@@ -32,11 +32,15 @@ def show_grid(grid):
 
 
 def show_moves(movable_tiles):
-    msg = 'You are allowed to move: '
+    return 'You are allowed to move (%s)' % ', '.join(str(tile) for tile in movable_tiles)
```
Comment (Kmaschta) [inline src/renderer/renderer.py]:
```diff
-str(tile) for tile in movable_tiles
+movable_tiles.map(str)
```

**Review by Kmaschta — CHANGES_REQUESTED** [review]

File: src/index.py [hunk 5]
```diff
@@ -1,44 +1,69 @@
...
 
-def router(action, grid, shuffled):
+def router(action, grid, started_grid, shuffled):
     if not shuffled and action in ['s', 'S']:
         print(shuffling())
-        return [shuffle(grid), True]
+        return [shuffle(grid), started_grid, True]
+    elif action in ['w', 'W']:
+        new_grid = change_size()
+        return [new_grid, new_grid.copy(), False]
```
Comment (Kmaschta) [inline src/index.py:20]: Not mandatory for this PR but in this kind of function, we prefer return an object with named attributes.

**Review by Kmaschta — APPROVED** [review]

**Pull request merged**